examples/todoapp/test/feature_todo_management/mock/todo_management_mock.dart

```diff
@@ -0,0 +1,74 @@
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+import 'package:rxdart/rxdart.dart';
+import 'package:todoapp/assets.dart';
+import 'package:todoapp/base/models/errors/error_model.dart';
+import 'package:todoapp/base/models/todo_model.dart';
+import 'package:todoapp/feature_todo_management/blocs/todo_management_bloc.dart';
+
+import 'todo_management_mock.mocks.dart';
+
+@GenerateMocks([
+  TodoManagementBlocStates,
+  TodoManagementBlocEvents,
+  TodoManagementBlocType
+])
+TodoManagementBlocType todoManagementMockFactory({
+  String? title,
+  String? description,
+  bool? showError,
+  bool? isLoading,
+  ErrorModel? errors,
+  TodoModel? onTodoSaved,
+}) {
+  final blocMock = MockTodoManagementBlocType();
+  final eventsMock = MockTodoManagementBlocEvents();
+  final statesMock = MockTodoManagementBlocStates();
+
+  when(blocMock.events).thenReturn(eventsMock);
+  when(blocMock.states).thenReturn(statesMock);
+
+  final titleState = title != null
+      ? title.length >= 3
```

Comment:
Avoid such logic in the mock factory. Try to be as explicit as possible. For example here you can add an additional argument to the `todoManagementMockFactory`